### PR TITLE
Make iOS compile against latest shared changes

### PR DESCRIPTION
**Pull request by newmskywalker**

### CODE REVIEW

File: shared/src/commonMain/kotlin/io.newm.shared/internal/implementations/WalletNFTTracksUseCaseImpl.kt [hunk 1]
```diff
@@ -38,11 +37,8 @@ internal class WalletNFTTracksUseCaseImpl(
         }
     }
 
-    @Throws(KMMException::class, CancellationException::class)
     override fun getNFTTrack(id: String): NFTTrack? {
-        return mapErrors {
```
Comment (martyu) [inline shared/src/commonMain/kotlin/io.newm.shared/internal/implementations/WalletNFTTracksUseCaseImpl.kt:43]: Whyd this get removed?

File: shared/src/commonMain/kotlin/io.newm.shared/public/usecases/WalletNFTTracksUseCase.kt [hunk 2]
```diff
@@ -57,7 +57,6 @@ interface WalletNFTTracksUseCase {
      * @param id The unique identifier of the NFT track to be retrieved.
      * @return NFTTrack? - The NFT track with the specified ID, or `null` if no such track is found.
      */
-    @Throws(KMMException::class, CancellationException::class)
```
Comment (martyu) [inline shared/src/commonMain/kotlin/io.newm.shared/public/usecases/WalletNFTTracksUseCase.kt:60]: Same here

**Review by martyu — CHANGES_REQUESTED** [review]

File: shared/src/commonMain/kotlin/io.newm.shared/public/usecases/WalletNFTTracksUseCase.kt (hunk 2 above)
Comment (newmskywalker) [reply shared/src/commonMain/kotlin/io.newm.shared/public/usecases/WalletNFTTracksUseCase.kt:60]: This fixes the error we were talking about on slack. 

image

It looks like throwing functions cannot return optional values. I looked at this specific function and it doesn't throw. It just returns null if the track is not found in the db. So I figured the easiest way to fix the issue was to make the function non-throwing. Let me know if that doesn't work!

Comment (newmskywalker) [reply shared/src/commonMain/kotlin/io.newm.shared/public/usecases/WalletNFTTracksUseCase.kt:60]: @martyu any additional thoughts on this PR?

**Review by cristhianescobar — DISMISSED** [review]

File: shared/src/commonMain/kotlin/io.newm.shared/public/usecases/WalletNFTTracksUseCase.kt (hunk 2 above)
Comment (cristhianescobar) [reply shared/src/commonMain/kotlin/io.newm.shared/public/usecases/WalletNFTTracksUseCase.kt:60]: @martyu are we good to merge this PR?

**Pull request merged**